qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -115,11 +118,11 @@ def __init__(
         estimator: BaseEstimator,
         fidelity: BaseStateFidelity,
         ansatz: QuantumCircuit,
-        optimizer: Optimizer | Minimizer,
+        optimizer: Union[Optimizer, List[Optimizer]] | Union[Minimizer, List[Minimizer]],
         *,
         k: int = 2,
         betas: Sequence[float] | None = None,
-        initial_point: Sequence[float] | None = None,
+        initial_point: Optional[Union[np.ndarray, List[np.ndarray]]] = None,
```

Comment:
I think the correct typehint here would be
```python
Sequence[float] | Sequence[Sequence[float]] | None
```
as 
* we're using the new way of writing typehints, without the `typing` module, so we wouldn't use `Optional` or `List`
* all algorithms currently use `Sequence[float]` over `np.ndarray` to be more general and also allow eg. lists of floats 🙂